kernel/work.c

```diff
@@ -644,34 +644,32 @@ static void work_queue_main(void *workq_ptr, void *p2, void *p3)
 
 		k_spin_unlock(&lock, key);
 
-		if (work != NULL) {
-			bool yield;
+		bool yield;
```

Comment:
nitpick, move this declaration to the beginning of the loop. Together with the others.

Reply:
Modified as suggested.